docs/data-sources/mongodb.md

```diff
@@ -0,0 +1,133 @@
+# MongoDB Data Source
+
+Connect your MongoDB database to liblab.ai and start building AI applications with your document data instantly.
+
+## Quick Start
+
+To connect MongoDB to liblab.ai, you need a MongoDB connection string in one of these formats:
+
+```
+mongodb://localhost:27017/database_name
+mongodb://username:password@localhost:27017/database_name
+mongodb+srv://username:password@cluster.mongodb.net/database_name
+```
+
+The system supports both local MongoDB instances and cloud services like MongoDB Atlas.
+
+## Connection Setup
+
+1. **Get your MongoDB connection string** from your MongoDB deployment
+2. **Add MongoDB as a data source** in liblab.ai
+3. **Enter your connection string** when prompted
+4. **Test the connection** to ensure it works properly
+
+### Supported Connection Formats
+
+- **Local MongoDB**: `mongodb://localhost:27017/database_name`
+- **MongoDB with authentication**: `mongodb://username:password@host:port/database_name`
+- **MongoDB Atlas**: `mongodb+srv://username:password@cluster.mongodb.net/database_name`
+
+## Supported Operations
+
+liblab.ai supports both read and write MongoDB operations:
+
+### Read Operations
+
+- **find**: Query documents with filters
+- **aggregate**: Complex data transformations and analytics
+
+### Write Operations
+
+- **insertOne**: Insert a single document
+- **insertMany**: Insert multiple documents
+- **updateOne**: Update a single document
+- **updateMany**: Update multiple documents
+- **deleteOne**: Delete a single document
+- **deleteMany**: Delete multiple documents
+
+## Query Structure
+
+All MongoDB queries in liblab.ai must be formatted as JSON objects with this structure:
+
+```json
+{
+  "collection": "collection_name",
+  "operation": "find",
+  "filter": { "status": "active" },
+  "options": { "limit": 10 }
+}
+```
+
+Required fields:
+
+- `collection`: The MongoDB collection name
+- `operation`: One of the supported operations
+
+Additional fields depend on the operation type:
+
+- **find**: `filter` (object), `options` (object)
```

Comment:
I am not sure we need all of these technical details concerning internal implementation for someone simply wanting to use the app and connect MongoDB and build apps. 
In a way, the liblab ai is there to abstract that away from the user.